kroxylicious-kms-provider-aws-kms/src/test/java/io/kroxylicious/kms/provider/aws/kms/AwsKmsEdekSerdeTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.aws.kms;
+
+import java.nio.ByteBuffer;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class AwsKmsEdekSerdeTest {
+
+    private final AwsKmsEdekSerde serde = new AwsKmsEdekSerde();
+
+    static Stream<Arguments> keyRefs() {
+        return Stream.of(
+                Arguments.of("keyref", "1234abcd-12ab-34cd-56ef-1234567890ab"));
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource(value = "keyRefs")
+    void shouldRoundTrip(String name, String keyRef) {
+        var edek = new AwsKmsEdek(keyRef, new byte[]{ 1, 2, 3 });
+        var buf = ByteBuffer.allocate(serde.sizeOf(edek));
+        serde.serialize(edek, buf);
+        buf.flip();
+        var deserialized = serde.deserialize(buf);
+        assertThat(deserialized).isEqualTo(edek);
+    }
+
+    static Stream<Arguments> sizeOf() {
+        return Stream.of(
+                Arguments.of(
```

Comment:
Do you have plans for more elements in the stream? Feels a bit odd have a single element stream why not just have it in the test case?

Reply:
Not for this PR, I'll remove the parameterised test for now.  Like I've said elsewhere, as a user of the Record Encryption with an AWS KMS, you should have the flexibility in the key identifiers you use (but later).

Reply:
No, it is a hang-over for a copy of the Vault implementation.